builtin/providers/vsphere/resource_vsphere_folder.go

```diff
@@ -0,0 +1,230 @@
+package vsphere
+
+import (
+	"fmt"
+	"log"
+	"path"
+	"strings"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/vmware/govmomi"
+	"github.com/vmware/govmomi/find"
+	"github.com/vmware/govmomi/object"
+	"golang.org/x/net/context"
+)
+
+type folder struct {
+	datacenter        string
+	existingPath      string
+	path              string
+}
+
+func resourceVSphereFolder() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceVSphereFolderCreate,
+		Read:   resourceVSphereFolderRead,
+		Delete: resourceVSphereFolderDelete,
+
+		Schema: map[string]*schema.Schema{
+			"datacenter": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+
+			"path": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"existing_path": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
```

Comment:
Is existing_path meant to be user settable? If not you should omit `Optional` so it's treated as read-only.